mock-recipe-server/api_index.html

```diff
@@ -0,0 +1,166 @@
+        Obtain a copy of Firefox with the SHIELD recipe client system add-on
+        installed. You can check <code>about:support</code> to ensure that you
+        have it.
+      </li>
+      <li>
+        Set the <code>extensions.shield-recipe-client.dev_mode</code> preference
+        to <code>true</code> to run recipes immediately on startup.
+      </li>
+      <li>
+        Set the <code>extensions.shield-recipe-client.logging.level</code>
+        preference to <code>0</code> to enable more logging.
+      </li>
+      <li>
+        Set the <code>security.content.signature.root_hash</code> preference to
+        <code>4C:35:B1:C3:E3:12:D9:55:E7:78:ED:D0:A7:E7:8A:38:83:04:EF:01:BF:FA:03:29:B2:46:9F:3C:C5:EC:36:04</code>.
+        This is the public hash used for verifying content signatures of the
+        test data.
+      </li>
+    </ol>
+    <h2>Verifying a Test Case</h2>
+    <p>
+      You have two options for verifying a test case depending on how thoroughly
+      you want to test the client:
+    </p>
+    <ul>
+      <li>
+        Find the test in the list below, set the
+        <code>extensions.shield-recipe-client.api_url</code> preference to the
+        given URL, and restart your browser. Upon restart, the client will
+        download recipes and execute them. This tests the recipe client from
+        startup to recipe execution.
+      </li>
+      <li>
+        If you have enabled the browser chrome debugging toolbox in the DevTools
+        settings, you can copy and paste the code from the textbox into the
+        Browser Console to immediately set the right preferences and fetch the
+        recipes. This skips testing a small amount of initialization, but is
+        otherwise a quick and reliable way of testing.
+      </li>
+    </ul>
+    <h2>Test Cases</h2>
+    {% for testcase in testcases %}
+      <div class="testcase" data-name="{{ testcase.name }}">
+        <h3>{{ testcase.name }}</h3>
+        <div class="testcase-content">
+          {{ testcase.description|safe }}
+          <p><strong>URL:</strong> <code class="url"></code></p>
+          <textarea class="browser-code"></textarea>
+        </div>
+      </div>
+    {% endfor %}
+    <script>
+      // Caluclate URLs relative to the current domain and populate each
+      // testcase with them.
+      let testcases = Array.from(document.querySelectorAll('.testcase'));
+      for (let testcase of testcases) {
+        let url = new URL(`/${testcase.dataset.name}/api/v1`, window.location);
+        testcase.querySelector('.url').textContent = url.href;
+        testcase.querySelector('.browser-code').value = (
+          `Cu.import('resource://shield-recipe-client/lib/Storage.jsm');\n` +
+          `Storage.clearAllStorage();\n` +
+          `Services.prefs.setCharPref("extensions.shield-recipe-client.api_url", "${url.href}");\n` +
+          `Cu.import('resource://shield-recipe-client/lib/RecipeRunner.jsm');\n` +
+          `RecipeRunner.start();`
```

Comment:
On the one hand, you can do multiline strings directly with backticks. On the other hand, the indentation is super annoying, so this is probably the right thing to do.